src/core/utils/hooks/useResizeObserver.tsx

```diff
@@ -6,29 +6,33 @@ import React from 'react';
 
 /**
  * Hook that uses `ResizeObserver` to return an element's size every time it updates.
- * @param elementRef ref of the element to observe resizes on.
+ * @param element the element to observe resizes on. You can pass `.current` member of the element's ref.
  * @returns stateful object containing height and width of the element.
+ *
+ * @example
+ * const elementRef = React.useRef<HTMLDivElement>(null);
+ * const { height, width } = useResizeObserver(elementRef.current);
+ * ...
+ * return <div ref={elementRef}>...</div>;
  */
-export const useResizeObserver = <T extends HTMLElement>(
-  elementRef: React.RefObject<T>,
-) => {
+export const useResizeObserver = <T extends HTMLElement>(element: T | null) => {
```

Comment:
Do you have an example with the issue of using ref?

Reply:
Yeah, it is the weird one which imodeljs users reported..
Basically it is connected with dynamic components, which rendered conditionally in some parent component, which has the ref. Firstly it can be null and then appear, which does not change the ref, but ref.current is not caught in our observer..

Reply:
> Lets change implementation to the same as useIntersection

Why? Current implementation makes more sense.